### PR TITLE
Remove android dependency

### PR DESCRIPTION
**Pull request by peter-tackage**
The Android dependency makes the build and CI more complex and slower than they really ought to be. I've migrated the Android Support Library annotations to be the Jetbrains equivalents, thus eliminating the Android dependency.

I've verified that in Android Studio and IntelliJ that it highlights a warning when `@Nullable` variable are used unchecked. Also according to their docs, SonarQube will report this too, as per: https://sonarqube.com/coding_rules#rule_key=squid%3AS2637

This reduces Travis-CI build time by 75% (4 mins -> 1 min)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-    - build-tools-23.0.1
-    - android-23
-    - extra-android-m2repository
-    - extra-google-m2repository
+language: java
+
+before_cache:
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+cache:
```
Comment (apoi) [inline .travis.yml:6]: Missing linebreak before

File: options/core/build.gradle [hunk 2]
```diff
@@ -35,19 +35,19 @@ sourceSets {
     test.java.srcDirs += 'src/test/kotlin'
 }
 
+
```
Comment (apoi) [inline options/core/build.gradle]: Extra linebreak

File: options/core/src/main/java/polanski/option/OptionUnsafe.java [hunk 3]
```diff
@@ -13,31 +13,31 @@ private OptionUnsafe() {
 
     /**
      * ATTENTION: Only use it when you know what you are doing!
-     *
+     * <p>
```
Comment (apoi) [inline options/core/src/main/java/polanski/option/OptionUnsafe.java]: Don't these have to be closed?

File: options/core/src/main/java/polanski/option/Some.java [hunk 4]
```diff
@@ -171,34 +171,34 @@ public OUT call(final IN it) {
...
-                    @Override
-                    public OUT call(final Object... list) {
-                        return f.call(combine(mValue, list));
-                    }
-                });
+            @Override
+            public OUT call(final Object... list) {
+                return f.call(combine(mValue, list));
+            }
+        });
```
Comment (apoi) [inline options/core/src/main/java/polanski/option/Some.java]: Could the indentation be improved? Looks a bit unnatural to decrease the indentation for the FuncN body

File: options/core/src/main/java/polanski/option/OptionUnsafe.java (hunk 3 above)
Comment (peter-tackage) [reply options/core/src/main/java/polanski/option/OptionUnsafe.java]: Looks like autoformat, I'll go through and fix up these unnecessary diffs.

File: options/core/build.gradle (hunk 2 above)
Comment (peter-tackage) [reply options/core/build.gradle]: It stole it from the `.travis.yml` file.

**Pull request merged**